LG AT IB auto drive for n seconds with x speed

LukaGrahek opened the pull request:
When the auto selector is set to one, the robot will run for x seconds at y left motor speed and z right motor speed.

LukaGrahek committed:
Added DriveWithTime command

added a command which will allow the robot to drive for n seconds in auto mode

LukaGrahek committed:
Commented DriveWithTime command

added comments

LukaGrahek committed:
Merge branch 'master' into lukalexAutoDriveForNseconds

LukaGrahek committed:
FINISHED drive for X seconds at N speed

Finished,  and tested the DriveWithTime command.

LukaGrahek committed:
Commented everything for the DriveWithSeconds command

added comments for the DriveWithSeconds command and everything associated with it

LukaGrahek committed:
Merge branch 'master' into lukalexAutoDriveForNseconds

Zinka010 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -75,7 +78,7 @@ private void configureButtonBindings() {
         
         // Instantiate the intake command and bind it 
-        intakeCommand = new OperatorIntakeCommand();
+       // intakeCommand = new OperatorIntakeCommand();
```
I know I did this for you, but if it's not being used, you should remove commented out code.

Zinka010 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +89,39 @@ private void configureButtonBindings() {
+         * When the selector is set to one, the robot will run for x seconds at y left motor speed and z right motor speed
+         */
+
+      // for the inputed variables: DriveWithTime(seconds (double), left motors speed (double), right motors speed (double))
```
This isn't really necessary since we can just `CTRL + CLICK` on the constructor.

Zinka010 commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+
+// initialization of the variables which will allow imported variables to be public(making usable throughout the DriveWithTime command).
+  public double seconds = 0;
+  public double leftSpeeds= 0;
```
Nitpick, add a space

Zinka010 commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+  }
+
+  //Initializes a timer
+  private Timer m_timer = new Timer();
```
1. Change the variable name to `Timer timer`
2. Nitpick: The initialization should probably go into the initialize method.

Zinka010 commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+  @Override
+  public void execute() {
+   
+    //sets motor speeds to those of which the user inputed
```
Nitpick: Change to `// Sets the motor speeds...`

Zinka010 commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+  //if using the withTimeout(); as the timer, return here.
```
Remove or indent this comment

Zinka010 reviewed (changes requested):
Seems great overall, just some minor formatting changes I'd like.

LukaGrahek committed:
Formatted comments and cleaned up code

Formatted comments and cleaned up code

georgetzavelas commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -78,6 +80,8 @@ public void robotPeriodic() {
         // and running subsystem periodic() methods.  This must be called from the robot's periodic
         // block in order for anything in the Command-based framework to work.
+
+        SmartDashboard.putNumber("AnalogSelector1", RobotContainer.analogSelectorOne.getIndex());
```
Is this test code?

LukaGrahek replied:
Yes, this was used to test if the selector switch on the robot was working and set to 1.
Do you want me to remove this code?

georgetzavelas replied:
Yes, if its test code we should remove it from the production level code unless we feel it will be used during competition

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -9,6 +9,7 @@
 
 import edu.wpi.first.wpilibj.GenericHID;
 import edu.wpi.first.wpilibj.Joystick;
+
```
Nit: Can we remove this line?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -41,13 +43,14 @@
     private Joystick driverStick;
     private Joystick controlStick;
-    private AnalogSelector analogSelectorOne;
+    public static AnalogSelector analogSelectorOne;
```
Any reasoning for making this public static?

LukaGrahek replied:
This is being used in the selector test code (line 84) in  Robot.java, which is why it's public.

georgetzavelas replied:
I think we should remove the debug statement from Robot.java. If you think it is needed maybe we can think of a more object oriented way to share this variable.

LukaGrahek replied:
I set it back to private.  It was previously public because it was being used for the test code which I now removed.

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -41,13 +43,14 @@
     private Command sensitiveDriverControlCommand;
     private Logger logger = Logger.getLogger("RobotContainer");
-    
+
```
Nit: Can we remove these extra lines?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -62,7 +65,7 @@ public RobotContainer() {
         configureButtonBindings();
     }
-    
+
```
Nit: Can we remove this extra line?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -72,11 +75,7 @@ public RobotContainer() {
-        // Instantiate the intake command and bind it 
-        intakeCommand = new OperatorIntakeCommand();
-        
+     
```
Any reason this code has been removed?

LukaGrahek replied:
Justin asked me to remove this code as it was not being used.

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +85,35 @@ private void configureButtonBindings() {
 
     }
-    
+
```
Nit: Can we remove this line?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +85,35 @@ private void configureButtonBindings() {
-        if (analogSelectorOne != null) selectorOne = analogSelectorOne.getIndex();
-        if (analogSelectorTwo != null) selectorTwo = analogSelectorTwo.getIndex();
+        int selectorOne = 1, selectorTwo = 1;
+        if (analogSelectorOne != null)
```
Nit: Your choice but consider adding {} around 1 line if statements across 2 lines. I know that in Java this is permitted but it is good practice to use brackets so that it becomes more obvious what is grouped with the if and for future additions it is simpler to add lines without cluttering changes for brackets.

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +85,35 @@ private void configureButtonBindings() {
-        
+
+
+        if (selectorOne == 1 || selectorTwo == 1) {
```
Should this be an `else if`?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +85,35 @@ private void configureButtonBindings() {
+
+        if (selectorOne == 1 || selectorTwo == 1) {
+ 
+            return new DriveWithTime(1.0, 0.2, 0.2);
```
These are magic numbers 😄 Can we make them into constants?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +85,35 @@ private void configureButtonBindings() {
         // Also return null if this ever gets to here because safety
         return null;
     }
+    
```
Nit: Can you remove these extra lines?

georgetzavelas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -86,26 +85,35 @@ private void configureButtonBindings() {
+    
+    
 }
+
```
Nit: Can you remove these extra lines?

georgetzavelas commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+ * @param rightSpeeds    when the motor is activated, this is the speed at which the RIGHT motors will be. speed can be inbetween -1 to 1.
+ *
+ */
+
```
Nitpick: can you remove this space so that the comments feel more associated with the code? In general this file seems to have a few too many extra empty lines.

georgetzavelas commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    // If using the withTimeout(); as the timer, return here.
```
Should we be stoping the timer here?

LukaGrahek replied:
No we shouldn't. We would only stop here if we were using the withTimeout(double seconds) command which counts downwards to 0, and  when it hits 0 it would interrupt the code and run whatever is in code is in here(the code would end the command). We switched to Timer() however, which counts up starting at 0. There were less problems while using Timer(). This code now isn't really needed, and it's just there in case we switch back to withTimeout(double seconds).

georgetzavelas replied:
Love this explanation! Thank you.

georgetzavelas commented on src/main/java/frc/robot/commands/DriveWithTime.java:
```diff
@@ -0,0 +1,79 @@
+public class DriveWithTime extends CommandBase {
+
+  // Initialization of the variables which will allow imported variables to be public(making usable throughout the DriveWithTime command).
+  public double seconds = 0;
```
Any reason these are public?

georgetzavelas reviewed (commented):
Great work guys! Some comments for you.

LukaGrahek committed:
Fixed spaces and some declarations

Fixed spaces and some declarations

LukaGrahek committed:
Fixed a space

Fixed a space

LukaGrahek committed:
Removed test code

Removed test code. The code checked what option the robots selectorswitch was on.

LukaGrahek committed:
Merge branch 'master' into lukalexAutoDriveForNseconds

Zinka010 reviewed (approved):
![vector oh yeah](https://media.tenor.com/images/69a26f90ebdcd3a6cfef615a48cad4ee/tenor.gif)

georgetzavelas reviewed (approved):
![hell-yeah-kid](https://user-images.githubusercontent.com/3611823/75096411-61461300-556d-11ea-9600-b050c7620d22.gif)

LukaGrahek commented:
![billNie](https://user-images.githubusercontent.com/55962832/75096751-9c961100-5570-11ea-9d54-ae1c8ee4e0b3.gif)
